crates/worker/src/partition/state_machine/tests/kill_cancel.rs

```diff
@@ -76,41 +83,102 @@ async fn kill_inboxed_invocation() -> anyhow::Result<()> {
     // assert that invocation status was removed
     assert!(let InvocationStatus::Free = current_invocation_status);
 
-    fn outbox_message_matcher(
-        caller_id: InvocationId,
-    ) -> impl Matcher<ActualT = restate_storage_api::outbox_table::OutboxMessage> {
-        pat!(
-            restate_storage_api::outbox_table::OutboxMessage::ServiceResponse(pat!(
-                restate_types::invocation::InvocationResponse {
-                    id: eq(caller_id),
-                    entry_index: eq(0),
-                    result: eq(ResponseResult::Failure(KILLED_INVOCATION_ERROR))
-                }
-            ))
-        )
-    }
-
     assert_that!(
         actions,
-        contains(pat!(Action::NewOutboxMessage {
-            message: outbox_message_matcher(caller_id)
-        }))
+        contains(
+            matchers::actions::invocation_response_to_partition_processor(
+                caller_id,
+                0,
+                eq(ResponseResult::Failure(KILLED_INVOCATION_ERROR))
+            )
+        )
     );
 
     let outbox_message = test_env.storage().get_next_outbox_message(0).await?;
 
     assert_that!(
         outbox_message,
-        some((ge(0), outbox_message_matcher(caller_id)))
+        some((
+            ge(0),
+            matchers::outbox::invocation_response_to_partition_processor(
+                caller_id,
+                0,
+                eq(ResponseResult::Failure(KILLED_INVOCATION_ERROR))
+            )
+        ))
     );
 
     test_env.shutdown().await;
     Ok(())
 }
 
-#[test(restate_core::test)]
-async fn kill_call_tree() -> anyhow::Result<()> {
-    let mut test_env = TestEnv::create().await;
+#[rstest]
+#[case(ExperimentalFeature::InvocationStatusKilled.into(), TerminationFlavor::Kill)]
+#[case(EnumSet::empty(), TerminationFlavor::Kill)]
```

Comment:
Maybe leave a comment explaining why `TerminationFlavor::Cancel` does not need to be tested for the experimental feature.